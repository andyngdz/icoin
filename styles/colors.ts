export const white = '#FFFFFF'
export const gray1 = '#333333'
