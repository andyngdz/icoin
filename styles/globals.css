@import url('//cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css');
@import url('//fonts.googleapis.com/css2?family=DM+Sans:ital,wght@0,400;0,500;0,700;1,400;1,500;1,700&display=swap');
@import url('/pace-theme.css');

body {
  font-family: 'DM+Sans', sans-serif;
}
