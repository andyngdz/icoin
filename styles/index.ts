import { createMuiTheme, responsiveFontSizes } from '@material-ui/core'
import Colors, { gray2, primary } from 'styles/colors'

const Theme = createMuiTheme({
  palette: {
    primary: {
      main: primary,
    },

    common: Colors,
  },

  typography: {
    fontFamily: "'Inter', sans-serif;",
    body1: { color: gray2 },
    button: { color: gray2 },
    subtitle2: { color: gray2 },
  },
})

export default responsiveFontSizes(Theme)
