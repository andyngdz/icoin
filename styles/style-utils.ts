import { teal } from '@material-ui/core/colors'

const StyleUtils = {
  createLinearGradient: (canvas: HTMLCanvasElement): CanvasGradient => {
    const ctx = canvas.getContext('2d')
    const gradient = ctx.createLinearGradient(0, 0, 0, canvas.height)
    gradient.addColorStop(1, teal[300])

    return gradient
  }
}

export { StyleUtils }
