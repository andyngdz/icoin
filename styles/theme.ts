import { createMuiTheme, responsiveFontSizes } from '@material-ui/core'
import { ICoinShadows, white, gray1 } from 'styles'
import { Shadows } from '@material-ui/core/styles/shadows'
import { teal, green } from '@material-ui/core/colors'
import { Theme } from '@material-ui/core'
import { TThemeMode } from 'types'

const ICoinTheme = (mode: TThemeMode): Theme => {
  const textPrimary = mode === 'dark' ? white : gray1

  return responsiveFontSizes(
    createMuiTheme({
      palette: {
        primary: teal,
        secondary: green,
        text: {
          primary: textPrimary
        }
      },

      shadows: ICoinShadows as Shadows,

      props: {
        MuiLink: {
          underline: 'none'
        }
      },

      typography: {
        fontFamily: "'Inter', sans-serif;"
      }
    })
  )
}

export { ICoinTheme }
