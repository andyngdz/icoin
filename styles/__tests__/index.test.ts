import { ICoinTheme } from 'styles'

describe('Theme', () => {
  it('Shuold return light theme config correctly', () =>
    expect(ICoinTheme('light')).toMatchSnapshot())

  it('Shuold return dark theme config correctly', () =>
    expect(ICoinTheme('dark')).toMatchSnapshot())
})
