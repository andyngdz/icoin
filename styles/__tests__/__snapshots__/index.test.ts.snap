// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Shuold return theme config correctly 1`] = `undefined`;
